charts/AxisScale.ts

```diff
@@ -76,26 +84,88 @@ export class AxisScale {
         return Math.min(this.range[1], this.range[0])
     }
 
-    getTickValues(): number[] {
-        const { scaleType, domain, d3_scale } = this
+    // When this is a log axis, only show so many grid lines because otherwise the chart would get
+    // too overwhelming. Different for mobile because screens are usually smaller.
+    @computed get maxLogLines() {
+        return isMobile() ? 8 : 10
+    }
+
+    getTickValues(): Tickmark[] {
+        const { scaleType, d3_scale, maxLogLines } = this
 
-        let ticks: number[] = []
+        let ticks: Tickmark[]
         if (scaleType === "log") {
-            let minPower10 = Math.ceil(Math.log(domain[0]) / Math.log(10))
-            if (!isFinite(minPower10)) minPower10 = 0
-            let maxPower10 = Math.floor(Math.log(domain[1]) / Math.log(10))
-            if (maxPower10 <= minPower10) maxPower10 += 1
-
-            for (let i = minPower10; i <= maxPower10; i++) {
-                ticks.push(Math.pow(10, i))
+            // This is a wild heuristic that decides how many tick lines and grid lines we want to
```

Comment:
I think this is a pretty neat heuristic. Clear description of the thought process.

Might it be worth moving this to a util method, with few dependencies? That way:

1. It would be easy to swap in and out different tick mark "strategies". We could have a storybook where you could view various charges and change the tick mark approach.
2. If you really nail it, other d3 users might use it.
3. Would be easier to write some unit tests.

Reply:
I think it can be neat _and_ wild at the same time :)
But yes, I wanted to nicely document the thought process and its implementation, because otherwise looking at it just one week later this is all bananas. And I quite like these daniel-style long form comments.

I also haven't yet looked at how other tools like d3, matplotlib or other plotting tools handle this case, because I would assume they need to do something similar as well. There might even be academic papers out on the topic.

And yes, re moving this to some kind of Util class, that might be an idea worth doing. I'll see if I find a find a good way of doing so next week.